### ISSUE TITLE
Babelrc resolving to wrong location

### ISSUE THREAD

**Issue by alansouzati** [storybooks/storybook#2635]
### Issue details

When I have a local babel preset, storybook is not able to locate it.

### Steps to reproduce

- Run `start-storybook` where you have a local babel preset. For example:

```
{
  "presets": [
    "./tools/babel-preset-env",
    "stage-2",
    "react"
  ]
}
```

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 3.3.3

### Affected platforms 

- Npm 5.6.0
- Node v8.9.4
- macOS High Sierra

### Screenshots / Screencast / Code Snippets (Optional)

> ERROR in ./storybook/addons.js
Module build failed: Error: Couldn't find preset "./tools/babel-preset-env" relative to directory "/Users/user/workspace/project/storybook"

**Comment by Hypnosphi** [comment]
Looks like it locates the file itself, but fails to resolve relative path from it correctly

Probably related: #2610

**Comment by alansouzati** [comment]
my current work around is to set a `.babelrc` file inside `storybook` folder that does not use the local preset.

I will try to investigate what is going on.

**Comment by dijs** [comment]
Any more word on this? I need to reference a local babel plugin and am running into the exact same issue.

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.20`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.20)
